### ISSUE TITLE
hypertimer is crashing node

### ISSUE THREAD

**Issue by erikvullings** [enmasseio/hypertimer#1]
Hi Jos,

I've been using hypertimer for the last couple of days, and it's very nicely documented and works smoothly, except for one annoying issue: it keeps crashing node. I only use the setTimeout method, and run in realtime mode (pace=1). The crash occurs in the finish() method, and the error is that the callback (that is called at the end) is a number > 0 so callback && callback() tries to run 1(), resulting in an exception.

Although I listen to the emitted error event, it still crashes node subsequently. 

To resolve my issue, I've added a check/hack (see below).

Am I doing something wrong?

``` javascript
    function finish() {
      // in case of an interval we have to reschedule on next cycle
      // interval must not be cleared while executing the callback
      if (timeout.type === TYPE.INTERVAL && current[timeout.id]) {
        timeout.occurrence++;
        timeout.time = timeout.firstTime + timeout.occurrence * timeout.interval;
        _queueTimeout(timeout);
        //console.log('queue timeout', timer.getTime().toISOString(), new Date(timeout.time).toISOString(), timeout.occurrence) // TODO: cleanup
      }

      // remove the timeout from the queue with timeouts in progress
      delete current[timeout.id];

    // My addition
      if (!isNaN(callback)) {
          if (callback !== 0) console.warn('Callback error: ' + callback);        
          return;
      }
      callback && callback();
    }
```

Cheers, Erik

**Comment by josdejong** [comment]
Thanks for your report. How can I reproduce this issue?

**Comment by erikvullings** [comment]
That's my problem too. It is not deterministic, and does not always occur.

**Comment by josdejong** [comment]
Hm. Any simple stress test that lets crash it sooner or later so we can output debug information?

**Comment by erikvullings** [comment]
I'll see if I can reproduce it in a separate application.

**Comment by josdejong** [comment]
That would be great!

**Comment by erikvullings** [comment]
OK, I was able to reproduce it: it actually occurs when I call the timer.setTimeout function with identical delay. If I call the same code, but just use setTimeout instead of timer.setTimeout, it works flawlessly. 

testtimer.js

``` javascript
var HyperTimer = require('hypertimer');
var timer = new HyperTimer();
for (var i = 0; i < 100; i++) {
    timer.setTimeout(function (x) {
        return function () {
            console.log(x);
        };
    }(i), 500);
}
```

This results in the following output (using the original code).

```
c:\dev>node testtimer.js
0
1
2
3
Error [TypeError: number is not a function]
c:\dev\web\cs\apps\RTI\website\node_modules\hypertimer\lib\hypertimer.js:551
      callback && callback();
                  ^
TypeError: number is not a function
    at finish (c:\dev\web\cs\apps\RTI\website\node_modules\hypertimer\lib\hypertimer.js:551:19)
    at _execTimeout (c:\dev\web\cs\apps\RTI\website\node_modules\hypertimer\lib\hypertimer.js:573:7)
    at Array.forEach (native)
    at onTimeout [as _onTimeout] (c:\dev\web\cs\apps\RTI\website\node_modules\hypertimer\lib\hypertimer.js:641:19)
    at Timer.listOnTimeout (timers.js:110:15)
```

**Comment by josdejong** [comment]
Ah, thanks I can reproduce the issue here. Lets see...

**Comment by josdejong** [comment]
I've published a bug fix release on npm (v2.1.2). See commit with the fix: https://github.com/enmasseio/hypertimer/commit/5bb962fdf283c7a313297eee744efbeebeedf2e2.

**Comment by erikvullings** [comment]
Thanks, I can confirm that this fixed my issues!
